src/resources/wifi_esp32.cc

```diff
@@ -156,36 +156,17 @@ class WifiResourceGroup : public ResourceGroup {
   ~WifiResourceGroup() {
 #if CONFIG_WPA_DEBUG_PRINT
     printf("[wifi] ~WifiResourceGroup()\n");
-    printf("[wifi] esp_wifi_deinit()\n");
-#endif
-    esp_err_t err = esp_wifi_deinit();
-#if CONFIG_WPA_DEBUG_PRINT
-    printf("[wifi] esp_wifi_deinit() -> done\n");
-#endif
-    if (err == ESP_ERR_WIFI_NOT_STOPPED) {
-#if CONFIG_WPA_DEBUG_PRINT
-      printf("[wifi] esp_wifi_stop()\n");
```

Comment:
Stopping is now taken care of when tearing down the `WifiEvents` resource. That resource only exists if we've tried to start the WiFi stack.